Source/Core/Core/HW/GCMemcard/GCMemcardDirectory.cpp

```diff
@@ -121,6 +122,61 @@ int GCMemcardDirectory::LoadGCI(const std::string& file_name, bool current_game_
   return NO_INDEX;
 }
 
+// This is only used by NetPlay but it made sense to put it here to keep the relevant code together
+std::vector<std::string> GCMemcardDirectory::GetFileNamesForGameID(const std::string& directory,
+                                                                   const std::string& game_id)
+{
+  std::vector<std::string> filenames;
+
+  u32 game_code = 0;
+  if (game_id.length() >= 4 && game_id != "00000000")
+    game_code = BE32((u8*)game_id.c_str());
+
+  std::vector<std::string> loaded_saves;
+  for (const std::string& file_name : Common::DoFileSearch({directory}, {".gci"}))
+  {
+    File::IOFile gci_file(file_name, "rb");
+    if (gci_file)
+    {
+      GCIFile gci;
+      gci.m_filename = file_name;
+      gci.m_dirty = false;
+      if (!gci_file.ReadBytes(&(gci.m_gci_header), DENTRY_SIZE))
+        continue;
+
+      const std::string gci_filename = gci.m_gci_header.GCI_FileName();
+      for (const std::string& loaded_save : loaded_saves)
```

Comment:
No prob. This can likely be changed into a `std::find_if` followed with

```cpp
if (iter != loaded_saves.end())
  continue;
```

where iter is the result of the `std::find_if`

Reply:
Could actually just be `std::find` I believe, as it's just a simple equality check and no further logic.

Reply:
Ah, yeah, that's true.